typescript-generator-gradle-plugin/src/main/java/cz/habarta/typescript/generator/gradle/TypeScriptGeneratorPlugin.java

```diff
@@ -14,7 +14,7 @@ public void apply(Project project) {
         final Task generateTsTask = project.task(Collections.singletonMap(Task.TASK_TYPE, GenerateTask.class), "generateTypeScript");
 
         for (Task task : project.getTasks()) {
-            if (task.getName().startsWith("compile") && !task.getName().startsWith("compileTest")) {
+            if ("compileClasspath".equals(task.getName())) {
```

Comment:
This needs to stay because otherwise `generateTypeScript` task would not depend on compilation tasks like `compileJava` or `compileKotlin` and could use stale classes.